components/Article/EditLink/index.tsx

```diff
@@ -0,0 +1,58 @@
+import { FaPencilAlt } from 'react-icons/fa';
+import { FormattedMessage } from 'react-intl';
+import styles from './index.module.scss';
+import { useLocale } from './../../../hooks/useLocale';
+import type { FC } from 'react';
+
+type EditLinkProps = {
+  absolutePath?: string;
+  relativePath?: string;
+  editPath?: string;
+};
+
+// TODO(HinataKah0): Change branch from major/website-redesign to main
+
+const baseEditURL =
+  'https://github.com/nodejs/nodejs.org/edit/major/website-redesign';
+
+const translationReadmeURL =
+  'https://github.com/nodejs/nodejs.org/blob/major/website-redesign/TRANSLATION.md';
+
+const EditLink: FC<EditLinkProps> = ({
+  absolutePath,
+  relativePath,
+  editPath,
+}) => {
+  const { currentLocale } = useLocale();
+
+  if (!relativePath && !editPath && !absolutePath) return null;
+
+  let href;
```

Comment:
Instead of making `let` you can outsource this whole thing to a function e.g: `getEditLink`.

The function returns an object with `{ translationKey, href }`, and inside the function, you have only a switch statement with `return` statements. (So no need for a `let`).

Also `+=` on the translationKey isn't a good approach honestly.

Reply:
About `+=`, are you referring to string concatenation performance issue?
Initially I was referring to [this](https://stackoverflow.com/questions/16696632/most-efficient-way-to-concatenate-strings-in-javascript). But there's a conflicting [here](https://moduscreate.com/blog/javascript-performance-tips-tricks/).

Actually I am not too sure myself. Do you mind suggesting?

Reply:
Done, thanks!

Initially I was a bit hesitant because of the small amount of logic but it looks cleaner now. Having `break` statement there is quite ugly.